src/pages/Lend/LoanSearchPage.vue

```diff
@@ -175,10 +186,16 @@ export default {
 				const activeChallengeHeaderExp = responseTeams[2];
 				const teamId = queryTeamId || userTeamId;
 
+				const lenderPublicId = args?.route?.query?.lender ?? '';
+				const shareChallengeFlag = args?.route?.query?.share_challenge ?? '';
+
 				if (teamId && activeChallengeHeaderExp) {
 					return Promise.all([
 						client.query({ query: teamsGoalsQuery, variables: { teamId, limit: 1 } }),
-						client.query({ query: TeamInfoFromId, variables: { team_id: teamId } })
+						client.query({ query: TeamInfoFromId, variables: { team_id: teamId } }),
+						shareChallengeFlag && lenderPublicId
```

Comment:
Having this outside the Promise.all?

Reply:
Should be fine here, since it has the `null` fallback.

Reply:
The `query` should be moved back to `Promise.all`